commerce/domain/inventory/cart/calculator/classes/InventoryCartCalculatorSample.cls

```diff
@@ -0,0 +1,60 @@
+/**
+  * @description 
+  * 
+  * USE CASE
+  * 
+  * The customer was notified by the warehouse that during the inventory process, SKU_RED_SHIRT was accidentally made available. 
+  * Due to a system issue, the warehouse is currently unable to mark it as unavailable again. 
+  * Customer does want to sell the SKU-RED_SHIRT in the store front
+  * 
+  * SI Solution
+  * Extended inventory calculator to show a message during checkout that SKU_RED_SHIRT that shipping migth be delay 
+  * 
+  */
+  public class InventoryCartCalculatorSample extends CartExtension.InventoryCartCalculator {
+
+
+    private static final String SKU_RED_SHIRT = 'SKU_RED_SHIRT'; 
+    private static final String SKU_RED_SHIRT_MESSAGE = 'Apologies for the inconvenience, but there may be a slight delay in the shipping of SKU_RED_SHIRT'; 
+
+    public InventoryCartCalculatorSample() {
+        super();
+    }
+
+     /**
+     * @description Constructor used by unit tests only. 
+     * @param apexExecutor Executor which executes various calculators. Can be used to stub calculation results or delegate calculations to actual Calculator. See <<CartCalculateExecutorMock>>.
+     */
+    public InventoryCartCalculatorSample(CartExtension.CartCalculateExecutorMock apexExecutor) {
+        // Must call super constructor in order for provided Executor to be used for calculations
+        super(apexExecutor);
+    }
+
+    public virtual override void calculate(CartExtension.CartCalculateCalculatorRequest request) { 
+
+        CartExtension.Cart cart = request.getCart();    
+
+        if (cart.getStatus() != CartExtension.CartStatusEnum.CHECKOUT) {
```

Comment:
I've tested you example on the real Storefront. And to make it work OOB (without other modification to the UI and CCA Orchestrator), I recommend to remove this condition. Let's allow the same logic in Checkout state as well? What do you think?
This would allow developer to copy-paste your sample into the store and immediately see the effect.